cypress/component/forms/cct/Cct.cy.tsx

```diff
@@ -0,0 +1,161 @@
+/// <reference types="cypress" />
+/// <reference path="../../../../cypress/support/index.d.ts" />
+import { mount } from "cypress/react18";
+import { MemoryRouter } from "react-router-dom";
+import { Provider } from "react-redux";
+import store from "../../../../redux/store/store";
+import { Cct } from "../../../../components/forms/cct/Cct";
+import {
+  openCctModal,
+  setCurrentProgEndDate,
+  setNewEndDates,
+  setProgName
+} from "../../../../redux/slices/cctCalcSlice";
+import dayjs from "dayjs";
+
+describe("Cct", () => {
+  it("renders the Cct component", () => {
+    const progName = "General Practice";
+    const endDate = dayjs().add(3, "year");
+    store.dispatch(setNewEndDates([]));
+    store.dispatch(openCctModal());
+    store.dispatch(setProgName(progName));
+    store.dispatch(setCurrentProgEndDate(endDate.format("YYYY-MM-DD")));
+
+    const newEndDate = (newPercent: number, currPercent: number) => {
+      const propStartDate = dayjs().add(16, "week").subtract(1, "day");
```

Comment:
Maybe this should default to the programme start date if this is more than 16 weeks in the future?